### DISCUSSION TITLE
Gremlin support for Graph modality

### DISCUSSION THREAD

**Discussion by kesavkolla** [unum-cloud/UStore#255] [Ideas]
Will ukv support Tinkerpop Gremlin for graph traversal? This will help to access the graph modalities in a uniform way.

**Comment by ashvardanian** [comment]
I have never heard of TinkerPop. Today we are targeting 2 extremes - Python and C, for maximal simplicity or performance. TinkerPop looks almost as complex as C and practically as slow as Python. Multi-threading in Java is very weak, and GPGPU is virtually non-existent. 

We will add TinkerPop to the evaluation list, and if wrong - we can add it in future versions, but for now the objective are leaning towards Nvidia Rapids and GPU-acceleration of graph processing, even if those are stored in external memory.
